etna/collections/models.py

```diff
@@ -124,6 +126,16 @@ def results_pages(self):
     ]
     subpage_types = ["collections.TopicExplorerPage", "collections.ResultsPage"]
 
+    def get_context(self, request):
+        context = super().get_context(request)
+        context["insights_topics"] = (
+            InsightsPage.objects.filter(topic=self)
+            .live()
+            .order_by("title")
+            .specific()[:3]
```

Comment:
You only need to use specific() when starting the queryset from the generic Page model manager, e.g. Page.objects.all(). This queryset will already be 'specific', because you're using the InsightsPage model manager.